File: androidApp/src/main/java/io/redlink/more/app/android/activities/studyStates/StudyPausedView.kt
/*
 * Copyright LBI-DHP and/or licensed to LBI-DHP under one or more
 * contributor license agreements (LBI-DHP: Ludwig Boltzmann Institute
 * for Digital Health and Prevention -- A research institute of the
 * Ludwig Boltzmann Gesellschaft, Österreichische Vereinigung zur
 * Förderung der wissenschaftlichen Forschung).
 * Licensed under the Apache 2.0 license with Commons Clause
 * (see https://www.apache.org/licenses/LICENSE-2.0 and
 * https://commonsclause.com/).
 */
package io.redlink.more.app.android.activities.studyStates

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import io.redlink.more.app.android.R
import io.redlink.more.app.android.extensions.getStringResource
import io.redlink.more.app.android.shared_composables.MediumTitle
import io.redlink.more.app.android.shared_composables.MoreBackground
import io.redlink.more.app.android.shared_composables.Title

@Composable
fun StudyPausedView() {
    MoreBackground() {
        Column(verticalArrangement = Arrangement.Center, horizontalAlignment = Alignment.CenterHorizontally, modifier = Modifier.fillMaxSize()) {
            Title(text = "${getStringResource(id = R.string.study_paused)}!", textAlign = TextAlign.Center)
            MediumTitle(text = "${getStringResource(id = R.string.study_paused_descr)}!", textAlign = TextAlign.Center)
        }
    }
}

@Preview
@Composable
fun StudyPausedPreview() {
    StudyPausedView()
}

File: iosApp/iosApp/Views/StudyStates/StudyPausedView.swift
//
//  StudyPausedView.swift
//  More
//
//  Created by Jan Cortiel on 25.07.23.
//  Copyright © 2023 Ludwig Boltzmann Institute for
//  Digital Health and Prevention - A research institute
//  of the Ludwig Boltzmann Gesellschaft,
//  Oesterreichische Vereinigung zur Foerderung
//  der wissenschaftlichen Forschung 
//  Licensed under the Apache 2.0 license with Commons Clause 
//  (see https://www.apache.org/licenses/LICENSE-2.0 and
//  https://commonsclause.com/).
//

import SwiftUI

struct StudyPausedView: View {
    private let stringsTable = "StudyStates"
    var body: some View {
        MoreMainBackgroundView {
            VStack(alignment: .center) {
                Spacer()
                Title(titleText: "\("Study currently paused".localize(withComment: "Study paused", useTable: stringsTable))!", textAlignment: .center)
                    .padding(.bottom, 8)
                Title2(titleText: "\("This study is currently paused by the Study Operator and will be resumed shortly".localize(withComment: "Study will be resumed shortly", useTable: stringsTable))!", textAlignment: .center)
                Spacer()
            }
        }
    }
}

struct StudyPausedView_Previews: PreviewProvider {
    static var previews: some View {
        StudyPausedView()
    }
}
